enumerable.rb

```diff
@@ -0,0 +1,108 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < self.size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    (self.size - 1).times do |i|
+      yield(self[i], i)
+    end
+    self
+  end
+
+  def my_select
+    item = []
+    self.my_each do |i|
+      item << i if yield(i)
+    end
+    item
+  end
+
+  def my_all?
+    item = []
+    self.my_each do |i|
+      item << i if yield(i)
+    end
+
+    return true if item.size == self.size
+    return
+  end
+
+  def my_any?
```

Comment:
* [ ] compare results from the custom method with the original
```
 p %w[ant bear cat].my_any?(/d/)                        #=> false
  p [nil, true, 99].my_any?(Integer)                     #=> true
  p [nil, true, 99].my_any?                              #=> true
  p [].my_any?
```
original method
```
  p %w[ant bear cat].any?(/d/)                        #=> false
  p [nil, true, 99].any?(Integer)                     #=> true
  p [nil, true, 99].any?                              #=> true
  p [].any?
```